### PR TITLE
OCPQE-20701 cpms qe cases

### PR DESCRIPTION
**Pull request by miyadav**
/pj-rehearse pull-ci-openshift-cluster-control-plane-machine-set-operator-main-regression-clusterinfra-azure-ipi-cpms-allcases

### CODE REVIEW

**Comment by miyadav** [comment]
/pj-rehearse pull-ci-openshift-cluster-control-plane-machine-set-operator-main-regression-clusterinfra-azure-ipi-cpms-allcases

**Comment by miyadav** [comment]
/pj-rehearse pull-ci-openshift-cluster-control-plane-machine-set-operator-main-regression-clusterinfra-azure-ipi-cpms-allcases

**Comment by miyadav** [comment]
@JoelSpeed @nrb @racheljpg ,PTAL when time permits , all tests passed and relevant CPMS cases are included , the failure seems to be platform related .
I think it is good to have these for the repo , as this regression is very time consuming later after merge.
Do let me know if you have any queries.
cc @sunzhaohua2 @huali9 @shellyyang1989 
```
`Writing JUnit report to /logs/artifacts/junit/junit_e2e_20240620-125353.xml

13 pass, 3 skip (5h59m57s)
+ set +x
try to handle result
/logs/artifacts/junit/junit_e2e_20240620-125353.xml
isJenkinsEnv: False\n
/logs/artifacts/junit/junit_e2e_2024-06-20-12-53-53.xml
isJenkinsEnv: False\n
done to handle result
success
Summarizing test results...
environment: line 400: row[2]: unbound variable
Exit trap triggered`
```

File: ci-operator/config/openshift/cluster-control-plane-machine-set-operator/openshift-cluster-control-plane-machine-set-operator-main.yaml [hunk 1]
```diff
@@ -269,6 +273,18 @@ tests:
         requests:
           cpu: 100m
     workflow: ipi-nutanix
+- as: regression-clusterinfra-azure-ipi-cpms-allcases
+  optional: true
+  steps:
+    cluster_profile: azure4
```
Comment (JoelSpeed) [inline ci-operator/config/openshift/cluster-control-plane-machine-set-operator/openshift-cluster-control-plane-machine-set-operator-main.yaml:279]: Are we picking Azure for a particular reason? Could this be AWS instead? Azure is one of the flakiest platforms, and AWS is more stable, but, if there's good reason to use Azure then happy to keep this way

Comment (miyadav) [reply ci-operator/config/openshift/cluster-control-plane-machine-set-operator/openshift-cluster-control-plane-machine-set-operator-main.yaml]: acked,changed to aws and adding for 4.15 and 4.16 as well.

**Comment by miyadav** [comment]
/pj-rehearse pull-ci-openshift-cluster-control-plane-machine-set-operator-release-4.15-regression-clusterinfra-aws-ipi-cpms-allcases

**Comment by miyadav** [comment]
/pj-rehearse pull-ci-openshift-cluster-control-plane-machine-set-operator-release-4.15-regression-clusterinfra-aws-ipi-cpms-allcases

**Comment by miyadav** [comment]
/pj-rehearse pull-ci-openshift-cluster-control-plane-machine-set-operator-release-4.15-regression-clusterinfra-aws-ipi-cpms-allcases

**Comment by miyadav** [comment]
/pj-rehearse pull-ci-openshift-cluster-control-plane-machine-set-operator-release-4.15-regression-clusterinfra-aws-ipi-cpms-allcases

**Comment by miyadav** [comment]
/pj-rehearse pull-ci-openshift-cluster-control-plane-machine-set-operator-release-4.15-regression-clusterinfra-aws-ipi-cpms-allcases

**Comment by JoelSpeed** [comment]
/pj-rehearse pull-ci-openshift-cluster-control-plane-machine-set-operator-main-regression-clusterinfra-aws-ipi-cpms-allcases

Would like to see the main run through as well

**Comment by miyadav** [comment]
@JoelSpeed QE tests have passed on main.

**Pull request merged**